modules/alloydb/variables.tf

```diff
@@ -277,6 +277,15 @@ variable "project_id" {
   type        = string
 }
 
+variable "psc_config" {
+  description = "PSC config for cluster and instance. Only one between network_config and psc_config can be used."
+  type = object({
+    enabled                   = optional(bool, null)
```

Comment:
We can remove this variable and either psc_config is null (then it is disabled), or not null - then enabled.

Though we need to make `psc_config` default to null, and then have more dancing around, when checking for `allowed_consumer_projects` as parent may be null.

Also - if in the end you want to keep current structure, it should be `optional(bool, false)`